File: backend/src/main/java/com/aivideo/canvas/entity/Asset.java
package com.aivideo.canvas.entity;

import jakarta.persistence.*;
import lombok.Data;

import java.time.LocalDateTime;

@Entity @Table(name = "assets") @Data
public class Asset {
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private Long userId;
    private Long projectId;
    private String fileName;
    private String fileType;
    private String mimeType;
    private Long fileSize;
    private String storageType;
    private String storagePath;
    private String fileUrl;
    private Integer width;
    private Integer height;
    private Double duration;
    @Column(columnDefinition = "JSON")
    private String metaJson;
    private LocalDateTime createdAt = LocalDateTime.now();
}

File: backend/app/schemas/asset.py
from datetime import datetime

from pydantic import BaseModel


class AssetData(BaseModel):
    id: int
    file_name: str
    file_type: str
    mime_type: str
    file_size: int
    file_url: str
    created_at: datetime

    class Config:
        from_attributes = True
